test/utils/session-core.test.js

```diff
@@ -0,0 +1,282 @@
+    const containerUrl = 'https://pod.example.com/Documents/';
+    const urlsToSet = [
+      containerUrl,
+      `${containerUrl}Bank%20Statement/`,
+      `${containerUrl}Passport/`,
+      `${containerUrl}Drivers%20License/`
+    ];
+
+    const webId = `https://${otherPodUsername}.${
+      mockSolidIdentityProvider.split('/')[2]
+    }/profile/card#me`;
+
+    urlsToSet.forEach(async (url) => {
+      await setDocAclForUser(session, url, 'update', webId, permissions);
+    });
+
+    expect(setDocAclForUser).toBeCalledTimes(4);
+  });
+});
+
+describe('checkContainerPermission', () => {
+  const otherPodUsername = 'pod2';
+  const documentsContainerUrl = `https://${otherPodUsername}.${
+    mockSolidIdentityProvider.split('/')[2]
+  }/Documents/`;
+  let catchExecuted = false;
+  const mockGetSolidDataset = vi.fn((url) => Promise.resolve(mockSolidDatasetFrom(url)));
+  const mockGetSolidDatasetRejected = vi.fn(() => Promise.reject(Error('dataset not found')));
+
+  beforeEach(() => {
+    session = {
+      fetch: vi.fn(),
+      info: {
+        webId: `${mockPodUrl}profile/card#me`
+      }
+    };
+  });
+  afterEach(() => {
+    vi.clearAllMocks();
+  });
+
+  it('expect try block to be ran if getSolidDataset resolves', async () => {
+    try {
+      await mockGetSolidDataset(documentsContainerUrl, { fetch: session.fetch });
+
+      expect(mockGetSolidDataset).toBeCalled();
+    } catch {
+      catchExecuted = true;
+    }
+
+    expect(catchExecuted).toBe(false);
+  });
+
+  it('expect catch block to be ran if rejected', async () => {
+    try {
+      await mockGetSolidDatasetRejected(documentsContainerUrl, { fetch: session.fetch });
+
+      expect(mockGetSolidDataset).toBeCalled();
+    } catch {
+      catchExecuted = true;
+    }
+
+    expect(catchExecuted).toBe(true);
+  });
+});
+
+describe('create container logic for Solid', () => {
+  const documentUrl = `${mockPodUrl}Documents/`;
+  let catchExecuted = false;
+  const mockGetSolidDataset = vi.fn((url) => Promise.resolve(mockSolidDatasetFrom(url)));
+  const createContainerAt = vi.fn((url) => Promise.resolve(mockSolidDatasetFrom(url)));
+  const mockGetSolidDatasetRejected = vi.fn(() => Promise.reject(Error('dataset not found')));
+
+  beforeEach(() => {
+    session = {
+      fetch: vi.fn()
+    };
+  });
+  afterEach(() => {
+    vi.clearAllMocks();
+  });
+
+  it('run try only if container exists', async () => {
+    try {
+      await mockGetSolidDataset(documentUrl, { fetch: session.fetch });
+      expect(mockGetSolidDataset).toBeCalledWith(
+        documentUrl,
+        expect.objectContaining({ fetch: session.fetch })
+      );
+    } catch {
+      catchExecuted = true;
+      await createContainerAt(documentUrl, { fetch: session.fetch });
+    }
+
+    expect(createContainerAt).not.toBeCalled();
+    expect(catchExecuted).toBe(false);
+  });
+
+  it('executes catch if getSolidDataset fails', async () => {
```

Comment:
This also doesn't test anything outside the test

Reply:
Removing test for this.